app/models/conference.rb

```diff
@@ -328,9 +328,10 @@ def event_distribution
   # ====Returns
   # * +hash+ -> hash
   def scheduled_event_distribution
-    confirmed_events = program.events.where(state: 'confirmed')
-    scheduled_value =  { 'value' => confirmed_events.where.not(start_time: nil).count, 'color' => 'green' }
-    unscheduled_value =  { 'value' => confirmed_events.where(start_time: nil).count, 'color' => 'red' }
+    confirmed_scheduled_events = Event.scheduled.select{|event| event.program == program}
+    confirmed_unscheduled_events = Event.unscheduled.select{|event| event.program == program}
```

Comment:
And here you could even do

`confirmed_unscheduled_events = program.events - confirmed_scheduled_event`

So you avoid calling the the reject(&:scheduled?) which would call the `scheduled?` method for all the objects. With that the `unscheduled` scope is not needed.